README.md

```diff
@@ -145,11 +145,17 @@ text.
 
 `options` - An optional object with the following keys:
 
-  * `ignoreEolWhitespace` - `true` to ignore any whitespace diffs at the end of
-    lines.
+  * `ignoreEolWhitespace` - `true` to ignore changes in whitespace at EOL.
+  * `ignoreChangeWhitespace` - `true` to ignore changes in amount of whitespace.
+    This ignores whitespace at line end, and considers all other sequences of
+    one or more whitespace characters to be equivalent.
+  * `ignoreWhitespace` - `true` to ignore whitespace when comparing lines.
+    This ignores differences even if one line has whitespace where the other line has none.
   * `useIndex` - `true` to compare against the index version instead of the HEAD
     version.
 
+NOTE: Only one of ignoreWhitespace, ignoreChangeWhitespace, ignoreEolWhitespace should be passed.
```

Comment:
I'd rather see this be a single option with an enum, something like `['None', 'Changed', 'EOL', 'All']`, and handle ignoreEolWhitespace as a legacy option. I think that's less confusing than 3 mutually-exclusive options.

Reply:
I'm not familiar with nan package, so I wasn't able to figure out how to do that. My original plan was to have all 3 booleans allowed, but it seems you've to first select arg by index, before checking the argument keyword.
Is it possible to do "if 'ignoreWhitespace' in args { doSomething; }" in nan? If you can point me to example code or documentation, I will update it.

Reply:
Actually, I misunderstood the code, this line doesn't make sense and multiple options will work together.

Reply:
Why has this been marked as resolved. I would suggest the same thing @50Wliu is suggesting to make it less confusing.

Reply:
As I said, you can pass all or none options, at it will work as expected. I added this line (which is removed now) by mistake.

Reply:
I am suggesting is having combining the white space variants to one option and have the values as enums.

```js
ignoreWhiteSpace: ['None', 'Changed', 'EOL', 'All']
```

Reply:
Ah, okay.
I personally prefer the current approach because it is similiar to how you pass these options to git diff. You can pass any switch and all switches, even though they're all supersets of the one before it.
That said, if you want, I can make it like you asked.

Reply:
I see your point now. What do you think about renaming `ignoreChangeWhitespace` to `ignoreWhitespaceChange` and ` ignoreWhitespace` to `ignoreWhitespaceAll`?

Reply:
How about making them as close to their corresponding git-diff switches, i.e. ignoreSpaceAtEOL, ignoreSpaceChange, ignoreAllSpace & keeping ignoreEolWhitespace for compatibility reasons?

Reply:
Sure. go for it.

Reply:
Done.